app/src/main/java/com/endava/myendava/adapters/ChipsAdapter.java

```diff
@@ -97,41 +143,90 @@ private String getTagGroupByPosition(int position) {
 
         TextView tagsGroupTextView;
         ChipGroup chipGroup;
+        TextView addButton;
+        ConstraintLayout layout;
 
         TagsViewHolder(@NonNull View itemView) {
             super(itemView);
+            layout = itemView.findViewById(R.id.layout);
+            addButton = itemView.findViewById(R.id.add_button);
             tagsGroupTextView = itemView.findViewById(R.id.tags_group_text_view);
             chipGroup = itemView.findViewById(R.id.tags_chip_group);
+            addButton.setOnClickListener(view -> {
+                Intent intent = new Intent(context, FilteredTagsActivity.class);
+                context.startActivity(intent);
+            });
+        }
+
+        public void removeChips() {
+            chipGroup.removeAllViews();
         }
 
         void createChip(final String tagGroup, final Tag tag, final OnChipClickedListener listener) {
             Chip chip = new Chip(chipGroup.getContext());
             chip.setText(tag.getTagName());
-            chip.setChipBackgroundColorResource(TagColorManager.getColor(tagGroup));
-            chip.setTextColor(context.getResources().getColor(android.R.color.white));
+            chip.setChipBackgroundColorResource(TagColorManager.getBackgroundColor(tagGroup));
+            int color = context.getColor(TagColorManager.getTextChipColor(tagGroup));
+            chip.setTextColor(color);
+            chip.setCloseIconEnabled(!"Project".equals(tagGroup) && isEditable);
+            chip.setOnCloseIconClickListener(view -> {
+                List<Tag> tags = getTagsByPosition(getAdapterPosition());
+                for (Tag tag1 : tags) {
```

Comment:
tag1 is not the best naming for a variable ... please find a better wording like currentTag or similar